mod-kb-ebsco-java/src/main/resources/domain/mod-kb-ebsco-java/features/kb-credentials.feature

```diff
@@ -3,155 +3,580 @@ Feature: KB Credentials
+    Given path '/eholdings/kb-credentials'
+    And request credentials
+    When method POST
+    Then status 201
+    * def id = response.id
+
+    Given path '/eholdings/kb-credentials', id
+    And request credentials
+    And set credentials.data.attributes.name = long_name
+    When method PUT
+    Then status 422
+    And match responseType == 'json'
+
+    Given path '/eholdings/kb-credentials', id
+    When method DELETE
+    And status 204
+
+  @Negative
+  Scenario: PUT /eholdings/kb-credentials/{id} should return status 422 and json if name is empty
+    Given path '/eholdings/kb-credentials'
+    And request credentials
+    When method POST
+    Then status 201
+    * def id = response.id
+
+    Given path '/eholdings/kb-credentials', id
+    And request credentials
+    And set credentials.data.attributes.name = ''
+    When method PUT
+    Then status 422
+    And match responseType == 'json'
+
+    Given path '/eholdings/kb-credentials', id
+    When method DELETE
+    And status 204
+
+  @Negative
+  Scenario: PUT /eholdings/kb-credentials/{id} should return status 422 and json if name is already exist
+    Given path '/eholdings/kb-credentials'
+    And request credentials
+    When method POST
+    Then status 201
+    * def id = response.id
+
+    Given path '/eholdings/kb-credentials', id
+    And request credentials
+    And set credentials.data.attributes.name = 'Dummy Credentials'
+    When method PUT
+    Then status 422
+    And match responseType == 'json'
+    And match response.errors[0].title == 'Duplicate name'
+
+    Given path '/eholdings/kb-credentials', id
+    When method DELETE
+    And status 204
+
+  @Negative
+  Scenario: PUT /eholdings/kb-credentials/{id} should return status 422 and json if url is empty
+    Given path '/eholdings/kb-credentials'
+    And request credentials
+    When method POST
+    Then status 201
+    * def id = response.id
+
+    Given path '/eholdings/kb-credentials', id
+    And request credentials
+    And set credentials.data.attributes.url = ''
+    When method PUT
+    Then status 422
+    And match responseType == 'json'
+
+    Given path '/eholdings/kb-credentials', id
+    When method DELETE
+    And status 204
+
+  @Negative
+  Scenario: PUT /eholdings/kb-credentials/{id} should return status 422 and json if customerId is empty
+    Given path '/eholdings/kb-credentials'
+    And request credentials
+    When method POST
+    Then status 201
+    * def id = response.id
+
+    Given path '/eholdings/kb-credentials', id
+    And request credentials
+    And set credentials.data.attributes.customerId = ''
+    When method PUT
+    Then status 422
+    And match responseType == 'json'
+
+    Given path '/eholdings/kb-credentials', id
+    When method DELETE
+    And status 204
+
+
+
+  @Negative
   Scenario: PATCH /eholdings/kb-credentials{id} should return status 400 and json if customerId invalid
```

Comment:
should return status 400, but in the scenario I don't see response with 400

Reply:
you are right, corrected